datafusion/common/src/join_type.rs

```diff
@@ -28,21 +28,26 @@ use crate::{DataFusionError, Result};
 /// Join type
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Hash)]
 pub enum JoinType {
-    /// Inner Join
+    /// Inner Join - Returns rows where there is a match in both tables.
     Inner,
-    /// Left Join
+    /// Left Join - Returns all rows from the left table and matching rows from the right table.
+    /// If no match, NULL values are returned for columns from the right table.
     Left,
-    /// Right Join
+    /// Right Join - Returns all rows from the right table and matching rows from the left table.
+    /// If no match, NULL values are returned for columns from the left table.
     Right,
-    /// Full Join
+    /// Full Join - Returns all rows when there is a match in either table.
+    /// Rows without a match in one table will have NULL values for columns from that table.
```

Comment:
> there is a match in either table

"a match" implies a matching row in left and right tables.

also  see above, this should also imply cardinality

maybe

```
/// Full Join - returns all matches between left and right table, 
/// plus left table rows without a match (with NULL values for right table columns)
/// plus right table rows without a match (with NULL values for left table columns)
```